consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -873,47 +873,45 @@ impl<
         // The enclave cannot provide a timestamp, so this happens in untrusted.
         signature.set_signed_at(chrono::Utc::now().timestamp() as u64);
 
-        BlockData::new(block, block_contents, Some(signature))
+        let metadata = BlockMetadata::new(
+            block.id.clone(),
+            Some((&self.scp_node.quorum_set()).into()),
+            self.enclave.get_ias_report().ok(),
+        );
+        // TODO: Is this the right key for signing the metadata?
+        let metadata =
+            SignedBlockMetadata::from_contents_and_keypair(metadata, &self.msg_signer_key)
+                .expect("Failed to sign block metadata");
```

Comment:
If we're signing the Block Metadata, this is definitely the unique message signing key that all consensus nodes are required to have (and send other peers to be considered part of the Quorum process). So if we're saying we're putting Block Metadata on chain signed by consensus nodes, I think this is definitely the key we want.

A subtle, yet significant TODO in this process is finding a mechanism to track a record of these signing keys. If we are forgetting the public key history of the consensus notes, it somewhat defeats the purpose of signing the block as we have no way to verify that these were the authoritative keys at a later point.

Reply:
#2104 tracks this